src/vec.rs

```diff
@@ -13,32 +13,65 @@ impl<T: PartialOrd> StalinFind<T> for Vec<T> {
     self.is_empty()
   }
 
-  fn stalin(&mut self, i: T, l: usize, r: usize) -> Option<usize>
-    where T: cmp::PartialEq + cmp::PartialOrd {
-    let m = (l + r) / 2;
-    if self[m] == i {
-      Some(m)
-    } else {
-      self.remove(m);
-      if m == 0 {
-        if self.is_empty() {
-          None
-        } else {
-          self.stalin(i, 0, self.len() - 1)
+    fn stalin(&mut self, i: T, l: usize, r: usize) -> Option<usize>
```

Comment:
you are not following original code formatting, I'm using two spaces and open bracket on same line... =_=

Reply:
sure